StortSpelProjekt/Engine/src/Renderer/TextManager.cpp

```diff
@@ -0,0 +1,274 @@
+#include "stdafx.h"
+#include <vector>
+#include "TextManager.h"
+
+#include "../Renderer/Texture/Texture.h"
+#include "../Renderer/DescriptorHeap.h"
+
+#include "../Misc/Window.h"
+#include "../Misc/AssetLoader.h"
+#include "../Misc/GUI2DElements/Text.h"
+#include "../Misc/GUI2DElements/Font.h"
+
+TextManager::TextManager()
+{
+}
+
+TextManager::~TextManager()
+{
+	for (auto textMap : m_TextMap)
+	{
+		delete textMap.second;
+	}
+	m_TextMap.clear();
+}
+
+std::map<std::string, TextData>* const TextManager::GetTextDataMap()
+{
+	return &m_TextDataMap;
+}
+
+TextData* TextManager::GetTextData(std::string name)
+{
+	return &m_TextDataMap[name];
+}
+
+void TextManager::AddText(std::string name)
+{
+	// Look if a font is chosen, otherwise, set the default font
+	if (m_pFont == nullptr)
+	{
+		m_pFont = AssetLoader::Get()->LoadFontFromFile(L"Arial.fnt");
+	}
+
+	// Default text
+	TextData textData = {};
+	textData.color = { 1.0f, 1.0f, 1.0f, 1.0f };
+	textData.padding = { 0.5f, 0.0f };
+	textData.pos = { 0.5f, 0.5f };
+	textData.scale = { 1.0f, 1.0f };
+	textData.text = L"DEFAULT_TEXT";
+
+	for (auto data : m_TextDataMap)
+	{
+		if (data.first == name)
+		{
+			Log::PrintSeverity(Log::Severity::WARNING, "It already exists a text with the name '%s'! Overwriting text data...\n", name.c_str());
+		}
+	}
+
+	m_TextDataMap[name] = textData;
+}
+
+void TextManager::UploadTextData(std::string name)
+{
+	Renderer::GetInstance().LoadTexture(m_pFont->GetTexture());
+
+	int numOfCharacters = GetNumOfCharacters(name);
+	auto textData = GetTextData(name);
+
+	Renderer* renderer = &Renderer::GetInstance();
```

Comment:
use renderer variable if you are using a variable. (line 65 you should get the renderer variable and use it.